pandas/core/indexes/api.py

```diff
@@ -214,7 +214,11 @@ def conv(i):
             return result.union_many(indexes[1:])
         else:
             for other in indexes[1:]:
-                result = result.union(other)
+                # GH 35092. Pass sort to Index.union
+                # Index.union expects sort=None instead of sort=True
+                if sort:
+                    sort = None
+                result = result.union(other, sort=sort)
```

Comment:
really? why is this

Reply:
IIRC sort is implemented for all union index types, or is it showing a warning?

Reply:
Ah, sorry, this confused me:

```
        sort : bool or None, default None
            Whether to sort the resulting Index.

            * None : Sort the result, except when

              1. `self` and `other` are equal.
              2. `self` or `other` has length 0.
              3. Some values in `self` or `other` cannot be compared.
                 A RuntimeWarning is issued in this case.

            * False : do not sort the result.
```
Works fine with `True`. Pushing the fix.

Reply:
hmm IIRC we did this deliberartly as we don't support this, can you see if can add an existing issue reference here (in both places if possible). future us will be puzzzled otherwise.

Reply:
@jreback 
Did a little digging. The API was changed in issue #24959. `sort` had been introduced to all the Index set-operations before, and then we discovered that the operations didn't guarantee sorting in edge cases. We decided to change the default to `sort=None` for set-ops to signify that and fix this in the future: ensure that `sort=True` always sorts for every set-op and then change the default to `sort=False` and deprecate `sort=None`. As far as I can tell, the last attempt to implement `sort=True` and deprecate `sort=None` was in #25980, and the PR went stale, so we never finished the job.

While we can and do hit edge cases in `Index.union`, there is a crutch in `_get_combined_index` that saves us if `union` doesn't sort:

```python
    if sort:
        try:
            index = index.sort_values()
        except TypeError:
            pass
```

I've added a comment to this PR to clarify all this stuff.